assets/app/scripts/controllers/build.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:PodController
+ * @description
+ * # ProjectController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('BuildController', function ($scope, $routeParams, DataService, project, BuildsService, $filter) {
+    $scope.build = null;
+    $scope.buildConfigName = $routeParams.buildconfig;
+    $scope.builds = {};
+    $scope.alerts = {};
+    $scope.renderOptions = $scope.renderOptions || {};    
+    $scope.renderOptions.hideFilterWidget = true;    
+    $scope.breadcrumbs = [
+      {
+        title: "Builds",
+        link: "project/" + $routeParams.project + "/browse/builds"
+      },
+      {
+        title: $routeParams.buildconfig,
+        link: "project/" + $routeParams.project + "/browse/builds/" + $routeParams.buildconfig
+      },      
+      {
+        title: $routeParams.build
+      }
+    ];
+
+    var watches = [];
+    var objectWatches = [];
+
+    project.get($routeParams.project).then(function(resp) {
+      angular.extend($scope, {
+        project: resp[0],
+        projectPromise: resp[1].projectPromise
+      });
+      DataService.get("builds", $routeParams.build, $scope).then(
+        // success
+        function(build) {
+          $scope.build = build;
+          var buildNumber = $filter("annotation")(build, "buildNumber");
+          if (buildNumber) {
+            $scope.breadcrumbs[2].title = "#" + buildNumber;
+          }
+
+          // If we found the item successfully, watch for changes on it
+          objectWatches.push(DataService.watchObject("builds", $routeParams.build, $scope, function(build, action) {
```

Comment:
I debated that, it would change the behavior of watch because we need to
explicitly know if something 404s.  Watch today doesn't return a promise
because it returns a handle for unwatch.  Also right now watchObject
doesn't just watch the object, it watches the whole resource type under the
covers.  I think we need to spend some time on data.js to make watchObject
better and to allow for filtering by labelFilter (so when you go to a
specific deployment config you dont have to watch ALL deployments).

On Thu, Sep 24, 2015 at 10:19 AM, Sam Padgett <notifications@github.com>
wrote:

> In assets/app/scripts/controllers/build.js
> <https://github.com/openshift/origin/pull/4726#discussion_r40323503>:
>
> > +    project.get($routeParams.project).then(function(resp) {
> > +      angular.extend($scope, {
> > +        project: resp[0],
> > +        projectPromise: resp[1].projectPromise
> > +      });
> > +      DataService.get("builds", $routeParams.build, $scope).then(
> > +        // success
> > +        function(build) {
> > +          $scope.build = build;
> > +          var buildNumber = $filter("annotation")(build, "buildNumber");
> > +          if (buildNumber) {
> > +            $scope.breadcrumbs[2].title = "#" + buildNumber;
> > +          }
> > +
> > +          // If we found the item successfully, watch for changes on it
> > +          objectWatches.push(DataService.watchObject("builds", $routeParams.build, $scope, function(build, action) {
>
> It would be nice if watchObject did the initial get for you.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4726/files#r40323503>.
>